### ISSUE TITLE
Method names should support symbol properties

### ISSUE THREAD

**Issue by slikts** [Microsoft/tslint-microsoft-contrib#459]
The current function-name rule fails when implementing an iterable:
```typescript
const it: Iterable<any> = {
  // [tslint] Method name does not match /^[a-z][\w\d]+$/: [Symbol.iterator] (function-name)
  [Symbol.iterator](): any {}
}
```
It should not fail; symbol properties and the iterable protocol are standard parts of JS since many years ago and should be supported.

**Comment by JoshuaKGoldberg** [comment]
Hi sklits 👋 thanks for the issue! You're absolutely right, this is a bug and should be fixed.

**Comment by elado** [comment]
here's my override:

```json
"function-name": [true, {
  "method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]+\\]?$",
  "private-method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]+\\]?$",
  "protected-method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]+\\]?$",
  "static-method-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]+\\]?$",
  "function-regex": "^\\*?\\[?[a-zA-Z][\\w\\d\\.]+\\]?$"
}],
```

**Comment by joelalejandro** [comment]
@JoshuaKGoldberg Is there a scheduled date for the `5.2.2` version of the package to include this fix? We're using this set of rules via `tslint-config-airbnb` and we're finding the same issue. We're applying a local override in the meantime, but it would be nice to see this go live :)

**Comment by JoshuaKGoldberg** [comment]
Thanks for bringing his up @joelalejandro - I'm planning on releasing a version after Hacktoberfest ends / early first week of November. It wasn't mentioned at all so I'll file a separate issue.
